python/orca/src/bigdl/orca/learn/pytorch/torch_runner.py

```diff
@@ -475,20 +475,27 @@ def _train_batch(self, batch, batch_info=None, callbacks=None):
                 calculate averages.
 
         """
-        # TODO: restore batch to what it should be.
-        self.call_hook(call_backs=callbacks, fn_name="before_train_iter")
         # unpack features into list to support multiple inputs model
+        # and restore batch to what it should be.
         features, target = batch
+        if torch.is_tensor(features):
+            self.batch = features, target
+        elif isinstance(features, (tuple, list)):
+            self.batch = *features, target
+        else:
+            invalidInputError(False,
+                              "Features should be tensor, list/tuple, "
+                              "but got {}".format(type(features)))
+        self.call_hook(call_backs=callbacks, fn_name="before_train_iter")
```

Comment:
call_backs is a bit strange? change to callbacks?

Reply:
sure